src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { FC, useState, useCallback, useMemo } from 'react';
+import debounce from 'lodash.debounce';
+import cn from 'classnames';
+
+import { peopleFromServer } from '../../data/people';
+import { Person } from '../../types/Person';
+import { filterPeople } from '../../utils/filterPeople';
+import { DropdownMenu } from '../DropdownMenu/DropdownMenu';
+
+type Props = {
+  selectedPerson: Person | null;
+  onSelect: React.Dispatch<React.SetStateAction<Person | null>>;
+  delay?: number;
+};
+
+export const Dropdown: FC<Props> = ({
+  selectedPerson,
+  onSelect,
+  delay = 300,
+}) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isActive, setIsActive] = useState(false);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const applyQuery = useCallback(debounce(setAppliedQuery, delay), []);
+
+  const filteredPeople = useMemo(
+    () => filterPeople(peopleFromServer, { appliedQuery }),
+    [appliedQuery],
+  );
+
+  const visiblePeople = selectedPerson ? peopleFromServer : filteredPeople;
+
+  const onInputFocus = () => {
+    setIsActive(true);
+  };
+
+  const onInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value.trimStart());
+    applyQuery(event.target.value);
+    onSelect(null);
+  };
+
+  const onSelectPerson = (person: Person) => {
+    onSelect(person);
+    setQuery(person.name);
+    setIsActive(false);
+  };
+
+  return (
+    <>
+      <div
+        className={cn('dropdown', {
+          'is-active': isActive,
+        })}
+      >
+        <div className="dropdown-trigger">
+          <input
+            type="text"
+            placeholder="Enter a part of the name"
+            data-cy="search-input"
+            className="input"
+            value={query}
+            onFocus={onInputFocus}
+            onChange={onInputChange}
+          />
+        </div>
+
+        {!!filteredPeople.length && (
+          <DropdownMenu
+            people={visiblePeople}
+            selectedPerson={selectedPerson}
+            handleSelectPerson={onSelectPerson}
```

Comment:
should be the other way round)
```suggestion
            onSelectPerson={handleSelectPerson}
```

Reply:
this was a correction from AI Buddy)